### PR TITLE
Update propolis with new Crucible Volume change

### PR DESCRIPTION
**Pull request by leftwo**
New crucible, which has updated Volume structure with Logger field. New VolumeConstructionRequest format which has sub-structures for all the fields.

This also adds support for using a new VCR for replacement
This enables downstairs replacement in a running Volume, and can also be used for migration to validate a VCR.

### CODE REVIEW

File: lib/propolis-client/src/handmade/api.rs [hunk 1]
```diff
@@ -16,7 +16,9 @@ use std::net::SocketAddr;
 use uuid::Uuid;
 
 // Re-export types that are of a public struct
-pub use crucible_client_types::VolumeConstructionRequest;
+pub use crucible_client_types::{
+    VcrFile, VcrRegion, VcrUrl, VcrVolume, VolumeConstructionRequest,
```
Comment (leftwo) [inline lib/propolis-client/src/handmade/api.rs:20]: I'm not 100% sure this was the correct place to export these, but it worked :)

**Comment by leftwo** [comment]
This adds some of the support required in https://github.com/oxidecomputer/propolis/issues/470

File: lib/propolis-client/src/handmade/api.rs (hunk 1 above)
Comment (leftwo) [reply lib/propolis-client/src/handmade/api.rs]: This change is no longer part of the PR

**Comment by leftwo** [comment]
This change has been mostly redacted.

All we have here now is an update to the Volume structure to include a `Logger` field, and support for the
few places that had wanted that logger as an arg.

**Review by gjcolombo — APPROVED** [review]

**Pull request merged**